[BUG] 2.4.6.1 build fails on ARM 64 (aarch64)

**Describe the bug**
Looks like a type conversion error specific to aarch64.

**Evidence**
There are many errors but here's one:
```
In file included from /builddir/build/BUILD/oiio-2.4.6.1/src/include/OpenImageIO/fmath.h:44,
                 from /builddir/build/BUILD/oiio-2.4.6.1/src/libutil/fmath.cpp:11:
/builddir/build/BUILD/oiio-2.4.6.1/src/include/OpenImageIO/simd.h: In member function 'void OpenImageIO_v2_4::simd::vbool4::load(bool, bool, bool, bool)':
/builddir/build/BUILD/oiio-2.4.6.1/src/include/OpenImageIO/simd.h:3264:31: note: use '-flax-vector-conversions' to permit conversions between vectors with differing element types or numbers of subparts
 3264 |     m_simd = vld1q_s32 (values);
      |                               ^
/builddir/build/BUILD/oiio-2.4.6.1/src/include/OpenImageIO/simd.h:3264:24: error: cannot convert 'int32x4_t' to 'OpenImageIO_v2_4::simd::vbool4::simd_t' {aka 'uint32x4_t'} in assignment
 3264 |     m_simd = vld1q_s32 (values);
      |              ~~~~~~~~~~^~~~~~~~
      |                        |
      |                        int32x4_t
```

Full build log here:
https://kojipkgs.fedoraproject.org//work/tasks/3501/95613501/build.log


**Platform information:**
 - OIIO branch/version: 2.4.6.1
 - OS: Fedora Rawhide
 - C++ compiler: gcc 12.2.1
 - Any non-default build flags when you build OIIO: (in build log)

--- a/src/include/OpenImageIO/simd.h
+++ b/src/include/OpenImageIO/simd.h
@@ -3261,7 +3261,9 @@ OIIO_FORCEINLINE void vbool4::load (bool a, bool b, bool c, bool d) {
     m_simd = _mm_castsi128_ps(_mm_set_epi32(-int(d), -int(c), -int(b), -int(a)));
 #elif OIIO_SIMD_NEON
     int values[4] = { -int(a), -int(b), -int(c), -int(d) };
-    m_simd = vld1q_s32 (values);
+    m_simd = vld1q_u32((const uint32_t*)values);
+    // this if we were using int:
+    // m_simd = vld1q_s32(values);
 #else
     m_val[0] = -int(a);
     m_val[1] = -int(b);
@@ -3501,7 +3503,9 @@ OIIO_FORCEINLINE bool extract (const vbool4& a) {
 #if OIIO_SIMD_SSE >= 4
     return _mm_extract_epi32(_mm_castps_si128(a.simd()), i);  // SSE4.1 only
 #elif OIIO_SIMD_NEON
-    return vgetq_lane_s32(a, i);
+    return vgetq_lane_u32(a, i);
+    // this if we were using int:
+    // return vgetq_lane_s32(a, i);
 #else
     return a[i];
 #endif
@@ -3514,8 +3518,11 @@ OIIO_FORCEINLINE vbool4 insert (const vbool4& a, bool val) {
     int ival = -int(val);
     return _mm_castsi128_ps (_mm_insert_epi32 (_mm_castps_si128(a), ival, i));
 #elif OIIO_SIMD_NEON
-    int ival = -int(val);
-    return vld1q_lane_s32(&ival, a, i);
+    uint32_t ival = uint32_t(val ? -1 : 0);
+    return vld1q_lane_u32(&ival, a, i);
+    // this if we were using int:
+    // int ival = -int(val);
+    // return vld1q_lane_s32(&ival, a, i);
 #else
     vbool4 tmp = a;
     tmp[i] = -int(val);

